lib/onix/product_supplies_methods.rb

```diff
@@ -22,22 +22,31 @@ module ProductSuppliesMethods
     #                :tax=>{:amount=>int, :rate_percent=>float}}]}]
     def supplies(keep_all_prices_dates = false)
       supplies = []
+      free_items = []
 
       # add territories if missing
       if self.product_supplies
         self.product_supplies.each do |product_supply|
           product_supply.supply_details.each do |supply_detail|
+
+            availability_date = supply_detail.availability_date
+            unless availability_date
+              if product_supply.availability_date
+                availability_date = product_supply.market_publishing_detail.availability_date
+              end
+            end
+            unless availability_date
+              if @publishing_detail
+                availability_date = @publishing_detail.publication_date
+              end
+            end
```

Comment:
Pourquoi mettre la même condition deux fois d'affilé ?
```suggestion
            unless availability_date
              if product_supply.availability_date
                availability_date = product_supply.market_publishing_detail.availability_date
              end
              if @publishing_detail
                availability_date = @publishing_detail.publication_date
              end
            end
```